src/app/devices/devices.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Observable } from 'rxjs-compat/Observable';
+import { AuthService } from '../core/auth.service';
+
+
+@Injectable()
+
+export class DevicesService {
+
+
+    constructor(private http: HttpClient, private authService: AuthService) {
+
+    }
+    getDevices(): Observable<Array<any>> {
+        return this.http.get<Array<any>>('api/devices');
+    }
+
+    getDevicesFiltered(device): Observable<Array<any>> {
+        const params = new HttpParams().set('search', device);
+        return this.http.get<Array<any>>('api/devices', { params: params });

```

Comment:
spoko ale można też prościej ;)
```
return this.http.get('api/devices', { params: {
    search: device
}});
```
jeżeli nie znamy typu to to: `<Array<any>>` jest nadmiarowe